src/main/java/gov/cms/madie/models/cqm/datacriteria/DiagnosisComponent.java

```diff
@@ -0,0 +1,23 @@
+package gov.cms.madie.models.cqm.datacriteria;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import gov.cms.madie.models.cqm.datacriteria.basetypes.Code;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+@Data
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DiagnosisComponent {
+  private String qdmVersion = "5.6";
+  private String _type = "QDM::DiagnosisComponent";
+  private String _id;
+  private Code code;
+  private Code presentOnAdmissionIndicator;
+  private int rank;
```

Comment:
looks good. have a question though- `int` type would be initialized to 0 by default which is a kind of valid rank. Should we use the wrapper type `private Integer rank` instead so that null would be allowed?

Reply:
@adongare Good point! I just changed it, thanks!